### PR TITLE
Added CommandEvent, deprecated (Remote)?ServerCommandEvent

### PR DESCRIPTION
**Pull request by SOF3**
## Introduction

Currently, plugins have to handle ServerCommandEvent and PlayerCommandPreprocessEvent separately to catch all commands by players and console/RCON. Not only is this inconvenient, but this also prevents plugins from catching events from other CommandSender types. This pull request provides a generic event `pocketmine\event\server\CommandEvent` that covers all CommandSender types.

### API changes

- Added `pocketmine\event\server\CommandEvent`
- Added an optional `$internal` parameter to `pocketmine\Server::dispatchCommand()`, which prevents `CommandEvent` from getting called
- Deprecated `pocketmine\event\server\{ServerCommandEvent, RemoteServerCommandEvent}`

### Behavioural changes

_nil_

## Backwards compatibility

Two events have been deprecated. This does not (yet) have any impact on backward compatibility.

## Follow-up

_nil_

## Tests

_nil_

### CODE REVIEW

**Review by dktapps — APPROVED** [review]
This has been one part of command handling that has irritated me for a long time.

I suggest that the deprecated events should be removed in 4.0.

**Comment by SOF3** [comment]
For those wondering: (Remote)?ServerCommandEvent is kept only for backward compatibility concerns. Updates and new plugins should switch to CommandEvent as of 3.2.0. Therefore, whether ServerCommandEvent or CommandEvent is called first will not be documented.

File: src/pocketmine/event/server/RemoteServerCommandEvent.php [hunk 1]
```diff
@@ -27,6 +27,8 @@
 
 /**
  * This event is called when a command is received over RCON.
+ *
+ * @deprecated Use CommandEvent instead.
```
Comment (dktapps) [inline src/pocketmine/event/server/RemoteServerCommandEvent.php]: will this trigger the appropriate deprecation warnings when a plugin registers a handler for this?

Comment (SOF3) [reply src/pocketmine/event/server/RemoteServerCommandEvent.php]: Not tested, but I think so

**Pull request merged**